spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/sql/TestStoragePartitionedJoins.java

```diff
@@ -133,7 +133,48 @@ public void removeTables() {
     sql("DROP TABLE IF EXISTS %s", tableName(OTHER_TABLE_NAME));
   }
 
-  // TODO: add tests for truncate transforms once SPARK-40295 is released
+  // TODO: SPJ does not currently leverage truncate(...) partition transforms for partition
+  // alignment.
+  // SPARK-40295 improved related areas, but full truncate support is tracked in SPARK-50593.
+  // This test documents current behavior; update/extend once SPARK-50593 lands.
+  @TestTemplate
+  public void testJoinWithTruncatePartitioning() {
+
```

Comment:
nit: remove empty line?

Reply:
Thanks for your suggestion, I will improve this code.